src/URIs.jl

```diff
@@ -457,15 +455,27 @@ function absuri(uri::URI, context::URI)
     @assert !isempty(context.host)
     @assert isempty(uri.port)
 
-    return merge(context; path=uri.path, query=uri.query)
+    return URI(context; path=uri.path, query=uri.query)
 end
 
+"""
+    joinpath(uri, path) -> URI
+
+Join the path component of URI and other parts.
+"""
+function Base.joinpath(uri::URI, parts::String...)
+    return URI(uri; path=join((uri.path, map(_trailing_first_slash, parts)...), '/'))
+end
+
+_trailing_first_slash(x::String) = startswith(x, '/') ? x[2:end] : x
```

Comment:
I see you've normalized cases like `"/a/b"` using `_trailing_first_slash`. I wonder if that's the behavior we want?

`joinpath` for normal paths treats these as absolute paths. For example...

```
julia> joinpath("a/b", "/c")
"/c"
```

Perhaps we should follow that instead.

Reply:
I see. I'm not very familiar with the URL standard, but I did this mainly because the REST API I wrap for defined these mapping in Python: https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/qiskit/providers/ibmq/api/rest/root.py#L31 so I thought this was a common way of defining identifiers, but I could be wrong.

Reply:
so people usually use path without slash to specific a relative location of URI as well and IBM people is doing something un-common then I guess. I realize following `joinpath` makes more sense indeed...